docs/sources/flow/reference/config-blocks/http.md

```diff
@@ -0,0 +1,148 @@
+  }
+}
+```
+
+## Arguments
+
+The `http` block supports no arguments and is configured completely through
+inner blocks.
+
+## Blocks
+
+The following blocks are supported inside the definition of `http`:
+
+Hierarchy | Block | Description | Required
+--------- | ----- | ----------- | --------
+tls | [tls][] | Define TLS settings for the HTTP server. | no
+
+[tls]: #tls-block
+
+### tls block
+
+The `tls` block configures TLS settings for the HTTP server.
+
+{{% admonition type="warning" %}}
+If the `tls` block is added and the configuration is reloaded when Grafana
+Agent is running, existing connections will continue to communicate over
+plaintext. Similarly, if the `tls` block is removed and the configuration is
+reloaded when Grafana Agent is running, existing connections will continue to
+communicate over TLS.
+
+To ensure all connections use TLS, start Grafana Agent with the `tls` block
+already configured.
+{{% /admonition %}}
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`cert_pem` | `string` | PEM data of the server TLS certificate. | `""` | conditionally
+`cert_file` | `string` | Path to the server TLS certificate on disk. | `""` | conditionally
+`key_pem` | `string` | PEM data of the server TLS key. | `""` | conditionally
+`key_file` | `string` | Path to the server TLS key on disk. | `""` | conditionally
+`client_ca_pem` | `string` | PEM data of the client CA to validate requests against. | `""` | no
+`client_ca_file` | `string` | Path to the client CA file on disk to validate requests against. | `""` | no
+`client_auth` | `string` | Client authentication to use. | `"NoClientCert"` | no
+`cipher_suites` | `list(string)` | Set of cipher suites to use. | `[]` | no
+`curve_preferences` | `list(string)` | Set of elliptic curves to use in a handshake. | `[]` | no
+`min_version` | `string` | Oldest TLS version to accept from clients. | `""` | no
+`max_version` | `string` | Newest TLS version to accept from clients. | `""` | no
+
+When the `tls` block is specified, arguments for the TLS certificate (using
+`cert_pem` or `cert_file`) and for the TLS key (using `key_pem` or `key_file`)
+are required.
+
+The following pairs of arguments are mutually exclusive, and only one may be
+configured at a time:
+
+* `cert_pem` and `cert_file`
+* `key_pem` and `key_file`
+* `client_ca_pem` and `client_ca_file`
+
+The `client_auth` argument determines whether to validate client certificates.
+The default value, `NoClientCert`, indicates that no client certificate
+validation is performed. The `client_ca_pem` and `client_ca_file` may only be
+configured when `client_auth` is not `NoClientCert`.
+
+The following values are accepted for `client_auth`:
+
+* `NoClientCert`: client certificates are neither requested nor verified.
+* `RequestClientCert`: requests clients to send an optional certificate.
+* `RequireAnyClientCert`: requires at least one certificate from clients that is not checked for validity.
```

Comment:
Does this mean that you MUST use a cert that has not been validated? Or that any N certs from the clients are allowed to be used without checking the validity?

Reply:
Made some changes in #4997, can we move this discussion over there?